packages/ethers5/exports/react.tsx

```diff
@@ -28,10 +28,10 @@ export function createWeb3Modal(options: Web3ModalOptions) {
 
 // -- Hooks -------------------------------------------------------------------
 export function useWeb3ModalProvider() {
-  const state = useSnapshot(EthersStoreUtil.state)
+  const { provider, providerType } = useSnapshot(EthersStoreUtil.state)
 
-  const walletProvider = state.provider as ethers.providers.ExternalProvider | undefined
-  const walletProviderType = state.providerType
+  const walletProvider = provider as ethers.providers.ExternalProvider | undefined
+  const walletProviderType = providerType
```

Comment:
Do we even need to reassign this ?

Reply:
Not sure, I was only playing with the code here while debugging cc @svenvoskamp what do you think?